### PR TITLE
Get rid of dependence on usercontent.riot.im

### PR DESCRIPTION
**Pull request by t3chguy**
Part of https://github.com/vector-im/riot-web/pull/12292

### CODE REVIEW

File: src/usercontent/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    const img = document.getElementById("img");
+    img.src = data.imgSrc;
+    img.style = data.imgStyle;
+
+    const a = document.getElementById("a");
+    a.style = data.style;
+}
+
+window.onmessage = function(e) {
+    if (lockOrigin === undefined || e.origin === lockOrigin) {
```
Comment (dbkr) [inline src/usercontent/index.js]: No need for backwards compat here so we could make the origin param mandatory.

**Comment by t3chguy** [comment]
> I assume making an object URL inside the sandboxed iframe, then passing it back to the main frame and using it as the src doesn't work?

browser doesn't let user click on it

> Also looks like this still requires the two clicks for decrypt/download? This is another thing that would be good to fix somehow.

indeed, will spin up an issue for this

**Comment by dbkr** [comment]
ok - fair enough, re-lgtm

**Comment by t3chguy** [comment]
https://github.com/vector-im/riot-web/issues/9035 tracks having to click twice

**Pull request merged**